soc/xtensa/intel_adsp/common/ipc.c

```diff
@@ -125,16 +132,20 @@ bool intel_adsp_ipc_is_complete(const struct device *dev)
 	return not_busy && !devdata->tx_ack_pending;
 }
 
-bool intel_adsp_ipc_send_message(const struct device *dev,
+int intel_adsp_ipc_send_message(const struct device *dev,
 			   uint32_t data, uint32_t ext_data)
 {
+	if (pm_device_state_is_locked(INTEL_ADSP_IPC_HOST_DEV))
```

Comment:
Thinking about it ... I don't think that is what you really want to check here ... What happens if the device in not locked but it is suspended ?? I think what you want to check here is that the device is active, so you should use `pm_device_state_get()`.

Reply:
Added check for device active state.